### PR TITLE
Add the Java 7 toolchain

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

**Review by jeffjohnson9046 — APPROVED** [review]

File: client/pom.xml [hunk 1]
```diff
@@ -20,6 +20,7 @@
     <properties>
         <maven.compiler.source>1.7</maven.compiler.source>
         <maven.compiler.target>1.7</maven.compiler.target>
+        <assertj.version>2.5.0</assertj.version>
```
Comment (jtreuting) [inline client/pom.xml:23]: Seems like this is unrelated and maybe just came along for the ride.  Is it needed?
